apps/openmw/mwrender/sky.cpp

```diff
@@ -1332,7 +1348,9 @@ class AlphaFader : public SceneUtil::StateSetUpdater
                         callback = callback->getNestedCallback();
                     }
 
-                    osg::ref_ptr<AlphaFader> alphaFader (new AlphaFader(mAlphaUpdate));
+                    FindParticleSystemVisitor fpart;
+                    node.accept(fpart);
+                    osg::ref_ptr<AlphaFader> alphaFader (new AlphaFader(mAlphaUpdate, fpart.mFound));
```

Comment:
This will only make one AlphaFader that will track *one* particle system that the node has (the last particle system child). Aren't all of them supposed to be tracked?

Can't NifOsg::FindByClassVisitor be used?

Reply:
> Can't NifOsg::FindByClassVisitor be used?

yes it could

Reply:
> Aren't all of them supposed to be tracked?


yes i will use a findbyname